client/etcd/config.go

```diff
@@ -26,16 +26,28 @@ import (
 	"github.com/m3db/m3x/instrument"
 )
 
-// ClusterConfig is the config for a zoned etcd cluster
+// ClusterConfig is the config for a zoned etcd cluster.
 type ClusterConfig struct {
-	Zone      string   `yaml:"zone"`
-	Endpoints []string `yaml:"endpoints"`
-	Cert      string   `yaml:"cert",omitempty`
-	Key       string   `yaml:"key",omitempty`
-	CA        string   `yaml:"ca",omitempty`
+	Zone      string     `yaml:"zone"`
+	Endpoints []string   `yaml:"endpoints"`
+	Auth      AuthConfig `yaml:"auth"`
```

Comment:
maybe make this type `*AuthConfig` to allow it to be nullable?

Reply:
The default struct can create a valid default option and saves a nil check, which is easier to use. Gonna keep it like this.